massa-deferred-calls/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+    pub fn get_total_gas(&self) -> u128 {
+        match self
+            .db
+            .read()
+            .get_cf(STATE_CF, DEFERRED_CALL_TOTAL_GAS.as_bytes().to_vec())
+            .expect(CRUD_ERROR)
+        {
+            Some(v) => {
+                let result = self
+                    .registry_changes_deserializer
+                    .total_gas_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(DEFERRED_CALL_DESER_ERROR)
+                    .1;
+                match result {
+                    SetOrKeep::Set(v) => v,
+                    SetOrKeep::Keep => 0,
+                }
+            }
+            None => 0,
+        }
+    }
+
+    pub fn put_entry(
+        &self,
+        slot: &Slot,
+        call_id: &DeferredCallId,
+        call: &DeferredCall,
+        batch: &mut DBBatch,
+    ) {
+        let mut buffer_id = Vec::new();
+        self.call_id_serializer
+            .serialize(call_id, &mut buffer_id)
+            .expect(DEFERRED_CALL_SER_ERROR);
+
+        let slot_bytes = slot.to_bytes_key();
+
+        let db = self.db.read();
+
+        // sender address
+        let mut temp_buffer = Vec::new();
+        self.call_serializer
+            .address_serializer
+            .serialize(&call.sender_address, &mut temp_buffer)
+            .expect(DEFERRED_CALL_SER_ERROR);
+        db.put_or_update_entry_value(
+            batch,
+            sender_address_key!(buffer_id, slot_bytes),
+            &temp_buffer,
+        );
+        temp_buffer.clear();
+
+        // target slot
+        self.call_serializer
+            .slot_serializer
+            .serialize(&call.target_slot, &mut temp_buffer)
+            .expect(DEFERRED_CALL_SER_ERROR);
+        db.put_or_update_entry_value(batch, target_slot_key!(buffer_id, slot_bytes), &temp_buffer);
+        temp_buffer.clear();
+
+        // target address
+        self.call_serializer
+            .address_serializer
+            .serialize(&call.target_address, &mut temp_buffer)
+            .expect(DEFERRED_CALL_SER_ERROR);
+        db.put_or_update_entry_value(
+            batch,
+            target_address_key!(buffer_id, slot_bytes),
+            &temp_buffer,
+        );
+        temp_buffer.clear();
+
+        // target function
+        self.call_serializer
+            .string_serializer
+            .serialize(&call.target_function, &mut temp_buffer)
+            .expect(DEFERRED_CALL_SER_ERROR);
+        db.put_or_update_entry_value(
+            batch,
+            target_function_key!(buffer_id, slot_bytes),
+            &temp_buffer,
+        );
+        temp_buffer.clear();
+
+        // parameters
+        self.call_serializer
+            .vec_u8_serializer
+            .serialize(&call.parameters, &mut temp_buffer)
+            .expect(DEFERRED_CALL_SER_ERROR);
+        db.put_or_update_entry_value(batch, parameters_key!(buffer_id, slot_bytes), &temp_buffer);
+        temp_buffer.clear();
```

Comment:
instead of trying not to forget to clear every time, we can just use a scope here for temp_buffer

Reply:
done